docs: cli instructions doc (#3356)
akuity/kargo: docs/docs/60-new-docs/50-user-guide/20-how-to-guides/05-installing-kargo-cli.md

---
sidebar_label: Installing the CLI
---

# Installing the Kargo CLI

> The Kargo CLI provides a command-line interface to manage Kargo efficiently.
> 
> The simplest and recommended way to install the CLI is via the <Hlt>CLI Tab</Hlt> in the Kargo UI.
> This ensures the Kargo CLI version matches the Kargo version installed in your cluster,
> making it easier to maintain compatibility and avoid issues.

Reviewer: ```suggestion
Although its web-based UI provides the best user experience, Kargo also offers a
command line interface (CLI) for users who may prefer it.

The easiest method of installing the CLI is to visit the Downloads page of the
UI, which can be found by selecting <Hlt>CLI</Hlt> tab from the left navbar.
There, you can select the appropriate binary for your operating system and
CPU architecture.

CLI downloads offered at this location will automatically match the version of
your Kargo API server for optimum compatibility.

Your downloaded binary will be suffixed with your OS and CPU architecture (e.g.
`kargo-darwin-arm64`), so be sure to rename it to simply `kargo` (or `kargo.exe`
for Windows).
```

> ![CLI Tab in Kargo UI](./img/cli-installation.png)

Reviewer: ```suggestion
    ![CLI Tab in Kargo UI](./img/cli-installation.png)
```

> _Alternatively_, if you prefer to install the CLI manually, you can run the following command:

Reviewer: ```suggestion
--------------------------------------------------------------------------------
If you prefer a more manual installation process, the following commands
will download the latest version of the CLI for your specific OS and CPU
architecture and will also rename the binary to `kargo` (or `kargo.exe`
on Windows.)
```

<Tabs groupId="os">
<TabItem value="mac-linux-wsl" label="Mac, Linux, or WSL" default>

```shell
arch=$(uname -m)
[ "$arch" = "x86_64" ] && arch=amd64
curl -L -o kargo https://github.com/akuity/kargo/releases/latest/download/kargo-"$(uname -s | tr '[:upper:]' '[:lower:]')-${arch}"
chmod +x kargo
```

</TabItem>
<TabItem value="windows" label="Windows Powershell">

```shell
Invoke-WebRequest -URI https://github.com/akuity/kargo/releases/latest/download/kargo-windows-amd64.exe -OutFile kargo.exe
```

</TabItem>

> </Tabs>

Reviewer: ~~~suggestion
    <Tabs groupId="os">
    <TabItem value="mac-linux-wsl" label="Mac, Linux, or WSL" default>

    ```shell
    arch=$(uname -m)
    [ "$arch" = "x86_64" ] && arch=amd64
    curl -L -o kargo https://github.com/akuity/kargo/releases/latest/download/kargo-"$(uname -s | tr '[:upper:]' '[:lower:]')-${arch}"
chmod +x kargo
    ```

    </TabItem>
    <TabItem value="windows" label="Windows Powershell">

    ```shell
    Invoke-WebRequest -URI https://github.com/akuity/kargo/releases/latest/download/kargo-windows-amd64.exe -OutFile kargo.exe
    ```

    </TabItem>
    </Tabs>
~~~

> After downloading, move the `kargo` binary (for Mac, Linux, or WSL) or `kargo.exe` (for Windows) to a

Reviewer: ```suggestion
1. Regardless of your chosen download method, complete the installation by
   moving the binary to a location in your file system that is included in the
   value of your `PATH` environment variable.
```

> After downloading, move the `kargo` binary (for Mac, Linux, or WSL) or `kargo.exe` (for Windows) to a
> folder included in your `PATH` environment variable. This makes it accessible from anywhere in the terminal.

Reviewer: ```suggestion
1. Regardless of your chosen download method, complete the installation by
   moving the binary to a location in your file system that is included in the
   value of your `PATH` environment variable.
```

---

> sidebar_label: Installing Kargo CLI

Reviewer: ```suggestion
sidebar_label: Installing the CLI
```

---

> # Installing Kargo CLI

Reviewer: ```suggestion
# Installing the Kargo CLI
```

> The Kargo CLI provides a command-line interface to manage Kargo efficiently, serving as
> a faster, automation-friendly alternative to the UI.

Reviewer: Two things here:

> The Kargo CLI provides a command-line interface

This seems like a redundant statement, so I think we should shorten this.

> automation-friendly alternative

This is less obvious, but I think we are best avoiding statements like this because of [this](https://docs.kargo.io/new-docs/faqs#how-do-i-integrate-kargo-into-my-ci-pipelines).

```suggestion
The `kargo` CLI is an alternative to Kargo's web-based UI for managing Kargo projects.
```

Choose the installation method for your operating system:

<Tabs groupId="os">
<TabItem value="mac-linux-wsl" label="Mac, Linux, or WSL" default>

1. Run the following command to download the CLI:
    ```shell
    arch=$(uname -m)
    [ "$arch" = "x86_64" ] && arch=amd64
    curl -L -o kargo https://github.com/akuity/kargo/releases/latest/download/kargo-"$(uname -s | tr '[:upper:]' '[:lower:]')-${arch}"
    chmod +x kargo
    ```

>     _Alternatively_, you can install the Kargo CLI through the <Hlt>CLI</Hlt> Tab in the Kargo UI:

Reviewer: imho, we should reverse the order of these. i.e. Promote the UI as the best option. Something particularly nice about that is the versions downloaded from the UI will match the version of Kargo that's installed in the cluster, which is a plus.

I also think it's easier.

Reviewer: On that note, I think we should _also_ update the quickstart to promote downloading from the UI. There, I think we can get away without even including the manual instructions because if the reader reaches that step of the quickstart, we know with certainty that the server side components of Kargo are installed and we even know the address of the download page.

Could you please bundle that change into this PR?

---
sidebar_label: Installing the CLI
---

# Installing the Kargo CLI

Although its web-based UI provides the best user experience, Kargo also offers a
command line interface (CLI) for users who may prefer it.

1. Download the CLI binary.

>     The easiest method of downloading the binary is by selecting

Reviewer: ```suggestion

    The easiest method of downloading the binary is by selecting
```

---
sidebar_label: Installing the CLI
---

# Installing the Kargo CLI

Although its web-based UI provides the best user experience, Kargo also offers a
command line interface (CLI) for users who may prefer it.

To install:

1. Download the CLI binary.

>     The easiest method of downloading the binary is by selecting

Reviewer: ```suggestion
    The easiest method of downloading the binary is by selecting the
```
